### ISSUE TITLE
Integration test with cookie does not see form

### ISSUE THREAD

**Issue by symbios-zi** [laravel/framework#16433]
Laravel 5.3
PHP7

Hello I am writing integration tests and trying to fill a form at order page. But the problem is that u can only open this page if you have basket_id cookie.

```
public function test_is_order_address_page_is_available_if_we_have_some_products_in_basket()
{
    $cookie = ['basket_id' => 'fdg435j4bmbhjnm'];
    $this->call('GET', '/order/address', [], $cookie);
    $this
        ->type('city', 'MyNewCity')
        ->type('name', 'My name')
        ->press('submit');
}
```
And I get an error ```Error: Call to a member function filter() on null```

I have read in documentation that this method works after ```$this->visit('/order')``` But I can't use this method coz this one cant uses cookies. Is there any solutions?

**Comment by GrahamCampbell** [comment]
What Laravel version please?

**Comment by symbios-zi** [comment]
@GrahamCampbell 
Laravel 5.3
PHP7

**Comment by GrahamCampbell** [comment]
Please provide the full version. You can find it by running `composer show -i`.

**Comment by symbios-zi** [comment]
@GrahamCampbell v5.3.22 The Laravel Framework.

I have found the solution:
```
$cookie = ['basket_id' => $basket->hash];

$this->makeRequest('GET', '/order/address', [], $cookie)
    ->select('1', 'region_id')
    ->type('Казань', 'city')
    ->type('123321', 'postcode')
    ->type('Ибрагимова','street')
    ->type('12','house')
    ->type('Пупкин','lastname')
    ->type('Вася','firstname')
    ->type('Петрович','patronymic')
    ->type('test@pupkin.com','email')
    ->press('submit_address_form');
```

But the problem is when I submit form using POST method, cookies does not available in store method. I can't test this form

**Comment by symbios-zi** [comment]
If we try to see into source files, we can notice that cookies is an empty array
https://github.com/laravel/framework/blob/3138e4071128c2721735e8e65621bd47b1960bc2/src/Illuminate/Foundation/Testing/Concerns/InteractsWithPages.php#L125

**Comment by themsaid** [comment]
@symbios-zi do you know a way of fixing this so that cookies are included with a form submission?

**Comment by symbios-zi** [comment]
@themsaid I changed native tests to Codeception. Its more flexible and powerful

**Comment by johnRivs** [comment]
@themsaid looking at the line of code @symbios-zi linked, I saw that a request to the form's action was being made passing an empty array as cookies. I tried editing that file so `press()` accepts the cookies as a second parameter so the cookies can reach the line where the request is being made. It didn't work, but seems to be for different reasons.

It turns out, if I do this (which is what ultimately said line of code does):
```php
$this->call('POST', 'register', [], ['foo' => 'bar']);
```
I get `null` in any middleware where I:
```php
dd($request->cookie('foo'));
```

It works just fine when done in the browser though. Any ideas?
